sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -174,6 +175,289 @@ def create_schema(
+        Implements the table or partition swap approach to insert-overwriting records.
+
+        Because this method executes multiple variants (full table replace, replace by time
+        range, replace by key, replace by partition), some upstream caller info is needed and
+        passed via kwargs.
+
+        Args:
+            table_name: Name of target table
+            source_queries: Source queries returning records to insert
+            columns_to_types: Column names and data types of target table
+            where: SQLGlot expression determining which target table rows should be overwritten
+            insert_overwrite_strategy_override: Not used by Clickhouse
+            kwargs:
+                dynamic_key: Key columns (replace by key only)
+                dynamic_key_exp: Expression to build key (replace by key only)
+                dynamic_key_unique: Whether more than one record can exist per key value (replace by key only)
+
+                is_inc_by_partition: Whether to overwrite partitions with only new records (incremental by partition only)
+
+        Returns:
+            Side effects only: execution of insert-overwrite operation.
+        """
+        target_table = exp.to_table(table_name)
+        columns_to_types = columns_to_types or self.columns(target_table)
+
+        temp_table = self._get_temp_table(target_table)
+        self._create_table_like(temp_table, target_table)
+
+        # REPLACE BY KEY: extract kwargs if present
+        dynamic_key = kwargs.get("dynamic_key")
+        if dynamic_key:
+            dynamic_key_exp = t.cast(exp.Expression, kwargs.get("dynamic_key_exp"))
+            dynamic_key_unique = t.cast(bool, kwargs.get("dynamic_key_unique"))
+
+        try:
+            # insert new records into temp table
+            for source_query in source_queries:
+                with source_query as query:
+                    # REPLACE BY KEY: if unique key, DISTINCTify by key columns so only one row is present per key
+                    if dynamic_key and dynamic_key_unique:
+                        query = query.distinct(*dynamic_key)  # type: ignore
+
+                    query = self._order_projections_and_filter(query, columns_to_types, where=where)
+                    self._insert_append_query(
+                        temp_table,
+                        query,
+                        columns_to_types=columns_to_types,
+                        order_projections=False,
+                    )
+
+            # REPLACE BY KEY: build `where` expression as "key IN (new rows' key values)"
+            if dynamic_key:
+                key_query = exp.select(dynamic_key_exp).from_(temp_table)
+                if not dynamic_key_unique:
+                    key_query = key_query.distinct()
+                where = dynamic_key_exp.isin(query=key_query)
+
+            # get target table partition key to confirm it's actually partitioned
+            table_partition_exp = self.fetchdf(
```

Comment:
Is this more like a `fetchone()` ? Since you just grab `[0]` and expect it to exist